salt/modules/yumpkg.py

```diff
@@ -3335,3 +3335,43 @@ def list_installed_patches(**kwargs):
         salt '*' pkg.list_installed_patches
     """
     return _get_patches(installed_only=True)
+
+
+def services_need_restarting(**kwargs):
+    """
+    .. versionadded:: NEXT
+
+    List services that use files which have been changed by the
+    package manager. It might be needed to restart them.
+
+    Requires systemd.
+
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt '*' pkg.services_need_restarting
+    """
+    if _yum() != "dnf":
+        raise CommandExecutionError("dnf is required to list outdated services.")
+    if not salt.utils.systemd.booted(__context__):
+        raise CommandExecutionError("systemd is required to list outdated services.")
+
+    cmd = ["dnf", "--quiet", "needs-restarting"]
+    dnf_output = __salt__["cmd.run_stdout"](cmd, python_shell=False)
+    if not dnf_output:
+        return []
+
+    services = set()
+    for line in dnf_output.split("\n"):
+        try:
+            pid, _ = line.split(":")
+        # Skip lines that can't be parsed
+        except ValueError:
+            continue
+        service = salt.utils.systemd.pid_to_service(pid.strip())
+        if service:
+            services.add(service)
```

Comment:
```suggestion
        pid, has_delim, _ = line.partition(':')
        if has_delim:
            service = salt.utils.systemd.pid_to_service(pid.strip())
            if service:
                services.add(service)
```

**suggestion:** this may be a bit cleaner way to write this - exception handling can be avoided, since partition will *always* return 3 values - and if there is no `:` then it won't even show up. The output may never have extra `:`, but this will also always limit to the first `:` per line. This also avoids having the `continue` in there as well.

Reply:
Agreed, that is a nicer way!